tslib/src/functions/check-validator/check-validator.ts

```diff
@@ -104,18 +95,23 @@ export class CheckValidatorV1 {
     }
   }
 
-  private validateCheckStructure (check: Record<string, unknown>): void {
-    const errorMessagePrefix = 'submitted check is missing the following properties:'
-    const missingProperties: string[] = []
-    for (let index = 0; index < requiredSubmittedCheckProperties.length; index++) {
-      const propertyName = requiredSubmittedCheckProperties[index]
-      if (!(propertyName in check)) {
-        missingProperties.push(propertyName)
+  private validateCheckStructureV2 (submittedCheck: any): void {
+    const validators = this.validatorProvider.getValidators()
+    const validationErrors: ICheckValidationError[] = []
+    for (let index = 0; index < validators.length; index++) {
+      const validator = validators[index]
+      const validationResult = validator.validate(submittedCheck)
+      if (validationResult !== undefined) {
+        validationErrors.push(validationResult)
       }
     }
-    const missingPropertyNames = missingProperties.join()
-    if (!RA.isEmptyArray(missingProperties)) {
-      throw new Error(`${errorMessagePrefix} ${missingPropertyNames}`)
+    if (validationErrors.length > 0) {
+      let validationErrorsMessage = `check validation failed. checkCode: ${submittedCheck.checkCode}`
```

Comment:
I am wondering how we would search this during live support.  I think we have a loose standard of the function name then a colon, e.g: `check-validator: validation for failed for checkCode 1234`

Reply:
the validators emit an error string, which is then consumed by the mail `check-validator.ts` file.  I have updated the main check validator to include the prefix, along with a list of each validation issue